### DISCUSSION TITLE
How to connect my app client to the temporal cluster?

### DISCUSSION THREAD

**Discussion by max-openline** [alexandrevilain/temporal-operator#644] [Q&A] [answered]
Hi everyone, first of all, thank you for this work

I deployed a temporal cluster and can see my pods running:
- worker pod
- fronted pod
- admin tools
- history pod
- matching
- UI
Now, I have an app that uses the temporal go SDK, and I must connect it to the `temporal server` which I do not see running?
Did I miss something? How can I establish the client connection without the server pod? (or should I manually deploy a temporal server?)

Cheers!

**Comment by alexandrevilain** [comment, marked as answer]
Hi @max-openline !

Your Go client should connect to the frontend pod, which distributes requests to others services (history & matching). 
So the server address is the frontend service on your cluster :)

**Comment by az-z** [comment]
@alexandrevilain , 
I hope you don't mind a followup question(s):
1. are services' names hardcoded? Do they follow some kind of naming schema? I have :
```
[az@dell5000 playground]$ kubectl get services -n demo
NAME                     TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)                      AGE
postgres                 ClusterIP   10.43.72.18    <none>        5432/TCP                     7d1h
prod-frontend            ClusterIP   10.43.117.37   <none>        7233/TCP,7243/TCP            3h27m
prod-frontend-headless   ClusterIP   None           <none>        9090/TCP,7233/TCP,6933/TCP   3h27m
prod-history-headless    ClusterIP   None           <none>        9090/TCP,7234/TCP,6934/TCP   3h27m
prod-matching-headless   ClusterIP   None           <none>        9090/TCP,7235/TCP,6935/TCP   3h27m
prod-ui                  ClusterIP   10.43.147.83   <none>        8080/TCP                     3h27m
prod-worker-headless     ClusterIP   None           <none>        9090/TCP,7239/TCP,6939/TCP   3h27m
```

**Reply by alexandrevilain** [reply]
Hi! 

Services’ name are following this pattern: {clusterName}-{temporal-service}.

In your example the cluster is named “prod”.

**Reply by az-z** [reply]
Thank you @alexandrevilain .
